File: packages/client/src/socket-path.ts
/**
 * Client-side socket path — daemon과 동일한 규칙으로 위치 결정.
 *
 * Linux/Mac: $XDG_RUNTIME_DIR/muxd.sock or /tmp/muxd-<uid>.sock
 * Windows: \\.\pipe\muxd-<user>
 *
 * muxd/daemon/socket-path.ts와 중복이지만 client가 muxd에 의존하지 않게 분리.
 * 규칙이 바뀌면 둘 다 갱신.
 */

import os from "node:os";
import path from "node:path";

export function defaultSocketPath(): string {
  if (process.platform === "win32") {
    const user = process.env.USERNAME || "default";
    return `\\\\.\\pipe\\muxd-${user.toLowerCase()}`;
  }
  const runtimeDir = process.env.XDG_RUNTIME_DIR;
  if (runtimeDir) return path.join(runtimeDir, "muxd.sock");
  const uid = process.getuid ? process.getuid() : 0;
  return path.join(os.tmpdir(), `muxd-${uid}.sock`);
}

File: packages/client-py/src/claude_mux/socket_path.py
"""
muxd daemon의 socket path resolver — TS daemon/socket-path.ts와 같은 규칙.

POSIX (Linux/Mac):
  - $XDG_RUNTIME_DIR/muxd.sock if set
  - else /tmp/muxd-<uid>.sock

Windows:
  - \\\\.\\pipe\\muxd-<username>

규칙 변경 시 양쪽(TS daemon + TS client + Python client) 모두 갱신.
"""

from __future__ import annotations

import os
import sys
import tempfile


def default_socket_path() -> str:
    if sys.platform == "win32":
        user = os.environ.get("USERNAME", "default").lower()
        return rf"\\.\pipe\muxd-{user}"
    runtime_dir = os.environ.get("XDG_RUNTIME_DIR")
    if runtime_dir:
        return os.path.join(runtime_dir, "muxd.sock")
    uid = os.getuid() if hasattr(os, "getuid") else 0
    return os.path.join(tempfile.gettempdir(), f"muxd-{uid}.sock")
